src/libstore/build/derivation-goal.cc

```diff
@@ -3391,9 +3406,12 @@ void DerivationGoal::registerOutputs()
         drvPathResolved = writeDerivation(worker.store, drv2);
     }
 
-    if (useDerivation || isCaFloating)
-        for (auto & [outputName, newInfo] : infos)
-            worker.store.linkDeriverToPath(drvPathResolved, outputName, newInfo.path);
+    if (settings.isExperimentalFeatureEnabled("ca-derivations") &&
+        (useDerivation || isCaFloating))
```

Comment:
Do we want to `registerDrvOutput` for non-CA derivations?

If so, we can do
```suggestion
    if (settings.isExperimentalFeatureEnabled("ca-derivations")
```
and make it unconditional if we like, since there no longer is a derivation foreign key.

If not, we could do
```suggestion
    if (isCaFloating)  {
        assert(settings.isExperimentalFeatureEnabled("ca-derivations"));
```

Reply:
Oh you're right, we can get rid of this